docs/resources/model_serving.md

```diff
@@ -0,0 +1,88 @@
+---
+subcategory: "Serving"
+---
+# databricks_model_serving Resource
+
+This resource allows you to manage [Model Serving Endpoints](https://docs.databricks.com/machine-learning/model-inference/serverless/serverless-real-time-inference.html) in Databricks.
+
+## Example Usage
+
+```hcl
+resource "databricks_model_serving" "this" {
+  name = "ads-serving-endpoint"
+  config {
+    served_models {
+      name = "prod_model"
+      model_name = "ads-model"
+      model_version = "2"
+      workload_size = "Small"
+      scale_to_zero_enabled = true
+    }
+    served_models {
+      name = "candidate_model"
+      model_name = "ads-model"
+      model_version = "4"
+      workload_size = "Small"
+      scale_to_zero_enabled = false
+    }
+    traffic_config {
+      routes {
+        served_model_name = "prod_model"
+        traffic_percentage = 90
+      }
+      routes {
+        served_model_name = "candidate_model"
+        traffic_percentage = 10
+      }
+    }
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the model serving endpoint. This field is required and must be unique across a workspace. An endpoint name can consist of alphanumeric characters, dashes, and underscores.
+* `config` - (Required) The model serving endpoint configuration.
+
+### config Configuration Block
+
+* `served_models` - (Required) Each block represents a served model for the endpoint to serve. A model serving endpoint can have up to 10 served models.
+* `traffic_config` - The traffic split configuration amongst the served models.
+
+### served_models Configuration Block
+
+* `name` - The name of a served model. It must be unique across an endpoint. If not specified, this field will default to -. A served model name can consist of alphanumeric characters, dashes, and underscores.
```

Comment:
@mehtayogita  can you confirm this is right? I thought the default endpoint served model name is `modelname-modelversion`

Reply:
Yes default is registeredModelName-modelVersion

Reply:
Ah we should probably fix the API docs then right: https://docs.databricks.com/dev-tools/api/latest/endpoints.html#tag/Serving-endpoints? Or is this not the source of truth anymore?

Reply:
Might as well fix it here, as it's manually maintained anyways - someone can follow up and fix the OpenAPI spec (and API docs) separately.